addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1308,221 +1308,276 @@ registry.background = SnippetOption.extend({
 
-        function _update(e) {
-            var posX = e.pageX - $(e.target).offset().left;
-            var posY = e.pageY - $(e.target).offset().top;
-            self.bgPos = {
-                x: clipValue(posX / $img.width() * 100).toFixed(2) + '%',
-                y: clipValue(posY / $img.height() * 100).toFixed(2) + '%',
-            };
-            self._updatePosInformation();
-            self._saveChanges();
+            this.$bgDragger.css('background-position', `${percentPosition.left}% ${percentPosition.top}%`);
+        };
+        $(this.ownerDocument).on('mousemove', drag);
+        $(this.ownerDocument).one('mouseup', () => {
+            this.$bgDragger.css({cursor: ''});
+            $(this.ownerDocument).off('mousemove', drag);
+        });
 
-            function clipValue(value) {
-                return Math.max(0, Math.min(value, 100));
-            }
+        function clamp(val, bounds) {
+            bounds = bounds.sort()
+            return Math.max(bounds[0], Math.min(val, bounds[1]));
         }
     },
     /**
-     * Removes all option-related classes and style on the target element.
+     * Initializes the overlay, binds events to the buttons, inserts it in the manipulators.
      *
      * @private
      */
-    _clean: function () {
-        this.$target.removeClass('o_bg_img_opt_contain o_bg_img_opt_custom o_bg_img_opt_repeat o_bg_img_opt_repeat_x o_bg_img_opt_repeat_y')
-                    .css({
-                        'background-size': '',
-                        'background-position': '',
-                    });
+    _initOverlay: function () {
+        this.$backgroundOverlay = $(qweb.render('web_editor.background_position_overlay'));
+        this.$overlayContent = this.$backgroundOverlay.find('.o_overlay_content');
+        this.$overlayBackground = this.$overlayContent.find('.o_overlay_background');
+
+        this.$backgroundOverlay.find('.o_btn_apply').on('click', () => {
+            this.$target.css('background-position', this.$bgDragger.css('background-position'));
+            this._toggleOverlay(false);
+        });
+        this.$backgroundOverlay.find('.o_btn_discard').on('click', () => this._toggleOverlay(false));
+
+        $('#oe_manipulators').append(this.$backgroundOverlay);
     },
     /**
-     * Restores the target style before last edition made with the option.
+     * Sets the overlay in the right place so that the draggable background
+     * renders over the target, and size the background item like the target.
      *
      * @private
      */
-    _discardChanges: function () {
-        this._clean();
-        if (this.previousState) {
-            this.$target.addClass(this.previousState[0]).css({
-                'background-size': this.previousState[1],
-                'background-position': this.previousState[2],
+    _dimensionOverlay: function () {
+        const $wrapwrap = $('#wrapwrap');
```

Comment:
The thing is, we want to darken anything that is not part of the editor's interface, in our case that is the wrapwrap, and it's not `:o_editable`. The highest ancestor which is, is the `#wrap`, but even its parent is only the `<main>` so covering either of those would leave the header and footer unobscured.

An alternative would be to use something like bootstrap's modal backdrop, which would cover everything, including the editor's interface, which might not be a bad thing.

Reply:
@sdegueldre Mmh ok, I guess we will wait for the editor team to come up with a great way to retrieve that wrapwrap element. Meanwhile you still can't suppose it exists. Use `document.body` and use its `width` and `height` instead of `innerWidth` and `innerHeight` and you should achieve the same result.

Reply:
As discussed, body isn't really an option because offset doesn't work with it, it has the wrong height, and parsing computedStyle paddings is brittle. Just noting it here for reference.